app/src/test/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentSheetControllerTest.kt

```diff
@@ -97,7 +98,7 @@ class ScheduleAppointmentSheetControllerTest {
 
     uiEvents.onNext(AppointmentDone)
 
-    verify(repository).schedule(patientUuid, date, Manual, facility)
+    verify(repository).schedule(eq(patientUuid), any(), eq(date), eq(Manual), eq(facility))
```

Comment:
This is a little tricky to do here since the property is an auto-generated `UUID`, and the only way to do an exact match is to ensure that we have control over the generation of the IDs themselves.

One option we have is to create an interface which is responsible for generating IDs and provide an implementation as an injection to all the controllers. This is out of scope for this PR, so I'll add a story for discussing this. What do you think?

Reply:
https://www.pivotaltracker.com/story/show/168315232

Reply:
What I meant was that we can pass `uuid` here when returning appointment:
```
whenever(repository.schedule(any(), any(), any(), any(), any())).thenReturn(Single.just(PatientMocker.appointment()))
```
And then use that `uuid` in verification. We have already started creating `UUID` in tests.
Does it make sense?

Reply:
The problem is that since this is the `ScheduleAppointmentSheetController`, unless we have a way to pass that `UUID` to the controller itself to use when scheduling an appointment, we can't use an `eq(appointmentUuid)` for verification. This is due to the fact that we use `UUID.randomUUID()` all over the codebase instead of having a replaceable component that will enable us to select the `UUID` that will be generated.

Reply:
This is also why I raised that Pivotal story so that we can discuss this and figure out what is the best approach for us to control `UUID` generation in tests.

Reply:
Oh. I understand what you mean.
Why are we passing `appointmentUuid` to `schedule()`?

Reply:
Because I needed a way to control the `UUID` that is being used to generate the appointment in the `schedule` method. Earlier, the repository was generating a random `UUID` internally which was causing problems while testing.

Reply:
Basically, the `schedule` function was impure since there was a source of randomness inside it. I needed to remove the impurity to make it easier to test. That, however, results in moving that impurity to one level above, i.e., the controller.

The overall solution is to stop using `UUID.randomUUID()` all over the code and replace it with a generator that we can control in tests.